master/buildbot/status/web/logs.py

```diff
@@ -210,8 +211,11 @@ def getChild(self, path, req):
             if path == log.getName():
                 if log.hasContents():
                     content = log.getText()
-                    if isinstance(log, HTMLLogFile) and ('xml-stylesheet' in content or 'nosetests' in content):
-                        return XMLTestResource(log, self.step_status)
+                    if isinstance(log, HTMLLogFile):
+                        if log.content_type == 'json':
+                            return JSONTestResource(log, self.step_status)
+                        elif log.content_type == 'xml' or 'xml-stylesheet' in content or 'nosetests' in content:
+                            return XMLTestResource(log, self.step_status)
                     else:
```

Comment:
I think if it doesnt match the 2 conditions above we want to still use the default IHTMLLog resource, the above changes doesnt seem to consider this case